crates/biome_js_analyze/src/lint/style/use_naming_convention.rs

```diff
@@ -386,20 +428,115 @@ declare_rule! {
-    ///   If the regular expression captures a part of the name, then this part is checked against `formats`.
-    ///   Only the first capture is tested. Other captures are ignored.
     /// - `formats`: the string [case] that the name must follow.
     ///   The supported cases are: [`PascalCase`], [`CONSTANT_CASE`], [`camelCase`], and [`snake_case`].
     ///
+    /// If both `match` and `formats` are set, then `formats` is checked against the first capture of the regular expression.
+    /// Only the first capture is tested. Other captures are ignored.
+    /// If nothing is captured, then `formats` is ignored.
+    ///
+    /// In the following example, we check the following conventions:
+    ///
+    /// - A private property starts with `_` and consists of at least two characters
+    /// - The captured name (the name without the leading `_`) is in [`camelCase`].
+    ///
+    /// ```json
+    /// {
+    ///     // ...
+    ///     "options": {
+    ///         "conventions": [
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember",
+    ///                     "modifiers": ["private"]
+    ///                 },
+    ///                 "match": "_(.+)",
+    ///                 "formats": ["camelCase"]
+    ///             }
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
     /// If `match` is set and `formats` is unset,
-    /// then the part of the name captured by the regular expression is forwarded to the next convention of the array.
+    /// then the part of the name captured by the regular expression is forwarded to the next conventions of the array.
+    /// In the following example, we require that private class members start with `_` and all class members are in ["camelCase"].
+    ///
+    /// ```json5
+    /// {
+    ///     // ...
+    ///     "options": {
+    ///         "conventions": [
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember",
+    ///                     "modifiers": ["private"]
+    ///                 },
+    ///                 "match": "_(.+)"
+    ///                 // We don't need to specify `formats` because the capture is forwarded to the next conventions.
+    ///             },
+    ///             {
+    ///                 "selector": {
+    ///                     "kind": "classMember"
+    ///                 },
+    ///                 "formats": ["camelCase"]
+    ///             }
+    ///         ]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// If a declaration is not selected or if a capture is forwarded while there is no more conventions,
+    /// then the declaration name is verified against the default conventions.
+    /// Because, the default conventions already ensure that class members are in ["camelCase"],
+    /// the previous example can be simplified to:
```

Comment:
```suggestion
    /// If a declaration is not selected or if a capture is forwarded while there are no more conventions,
    /// then the declaration name is verified against the default conventions.
    /// Because the default conventions already ensure that class members are in ["camelCase"],
    /// the previous example can be simplified to:
```